rust/Cargo.toml

```diff
@@ -33,7 +33,7 @@ reqwest = { version = "0.11", default-features = false, features = ["rustls-tls"
 
 # Azure
 azure_core = { git = "https://github.com/Azure/azure-sdk-for-rust", optional = true, rev = "2f2caff49b3e1ba08f430f82e16994670dddda95" }
-azure_storage = { git = "https://github.com/Azure/azure-sdk-for-rust", optional = true, rev = "2f2caff49b3e1ba08f430f82e16994670dddda95", features = ["blob", "account"] }
+azure_storage = { git = "https://github.com/Azure/azure-sdk-for-rust", optional = true, rev = "2f2caff49b3e1ba08f430f82e16994670dddda95", features = ["account", "blob", "data_lake"] }
```

Comment:
it doesn't appear that the `data_lake` feature is being used so can it be removed from the pull request, or is there still more work upstream coming to support this

Reply:
Thanks @rtyler. Sorry, went on vacation :) The goal of this PR is to switch everything to the ADLS Gen2 API. Upstream is ready enough at this point, but there will be more upstream changes coming that we can adopt in delta-rs later on.